runtime/sema/check_remove_statement.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2022 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sema
+
+import (
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/common"
+)
+
+func (checker *Checker) VisitRemoveStatement(statement *ast.RemoveStatement) (_ struct{}) {
+	nominalType := checker.convertNominalType(statement.Attachment)
+	base := checker.VisitExpression(statement.Value, nil)
+
+	if nominalType == InvalidType {
+		return
+	}
+
+	attachmentType, isCompositeType := nominalType.(*CompositeType)
+
+	if !isCompositeType || attachmentType.Kind != common.CompositeKindAttachment {
+		checker.report(
+			&RemoveFromInvalidTypeError{
```

Comment:
Should this be a different error compared to the below ones? i.e: Here it is the attachment type that is wrong. More like "removing a non-attachment type error"

Reply:
This is what the error message looks like in practice, but I will rename the error type to be more representative of the fact that it handles both cases.